src/app/features/home/custom-camera/custom-camera.page.ts

```diff
@@ -200,11 +205,53 @@ export class CustomCameraPage implements OnInit, OnDestroy {
   }
 
   async startPreviewCamera() {
+    if ((await this.canStartPreviewCamera()) === false) {
+      return;
+    }
+
     await this.customCameraService.startPreviewCamera();
     await this.customCameraService.setCameraQuality(this.cameraQuality$.value);
     await this.syncCameraState();
   }
 
+  async canStartPreviewCamera() {
+    try {
+      const permissions = await this.customCameraService.requestPermissions();
+      if (
+        permissions.camera !== 'granted' ||
+        permissions.microphone !== 'granted'
```

Comment:
In current published app version do we always require the user to grant microphone permission in order to use camera? (even if user doesn't want to use video recording?)

Reply:
@shc261392 to keep logic simple I ask all permissions at once. 

I will ask QA if we need lazy permission feature (aka ask microphone permission when user clicks record video only).